engine/orchestration/src/com/cloud/agent/manager/AgentManagerImpl.java

```diff
@@ -821,9 +819,9 @@ protected boolean handleDisconnectWithInvestigation(final AgentAttache attache,
             try {
                 nextStatus = host.getStatus().getNextStatus(event);
             } catch (final NoTransitionException ne) {
-                /* Agent may be currently in status of Down, Alert, Removed, namely there is no next status for some events.
-                 * Why this can happen? Ask God not me. I hate there was no piece of comment for code handling race condition.
-                 * God knew what race condition the code dealt with!
+                /*
```

Comment:
@GabrielBrascher and @rafaelweingartner Although I appreciate your remarks (especially the one about refactoring a method out of a bigger method), I'm not sure if you realise I've just formatted this class, and fixed a few logging messages. Therefore I won't really be maintaining the rest of the functionality. I do encourage you two to make a PR with your improvements in it. This is after all open-source and cooperation based.